lib/java/com/google/android/material/textfield/TextInputLayout.java

```diff
@@ -798,7 +798,7 @@ public EditText getEditText() {
   @Override
   public int getBaseline() {
     if (editText != null) {
-      return editText.getPaddingTop() + editText.getBaseline();
+      return editText.getBaseline() + getPaddingTop() + calculateLabelMarginTop();
```

Comment:
Looking good, but boxBottomOffsetPx still needs to be accounted for. It should be added to the calculation only if boxBackgroundMode != BOX_BACKGROUND_NONE. You may want to split that logic out into a separate method.

Reply:
#### I'm not sure that's right. Here's a screenshot _without_ the extra padding:
![screenshot_20180904-153231](https://user-images.githubusercontent.com/9490724/45061283-ab28f980-b058-11e8-88fd-334968f3a7ee.png)

#### And here is _with_ the extra padding:
![screenshot_20180904-153407](https://user-images.githubusercontent.com/9490724/45061288-b4b26180-b058-11e8-9b16-a83132987746.png)


BTW, here's my code:
```java
int padding = getPaddingTop();
if (boxBackgroundMode != BOX_BACKGROUND_NONE) {
  padding += boxCollapsedPaddingTopPx;
}
```

Reply:
The extra offset to account for is boxBottomOffsetPx, not boxCollapsedPaddingTopPx.

Reply:
Nope, that will move the legacy widget down. The first screenshot was fine... what is there to fix?

Reply:
It shouldn't move the legacy widget down if the offset is only being added when boxBackgroundMode != BOX_BACKGROUND_NONE.

From what I can see in the first screenshot, the legacy text field is a little off. Can you add a screenshot that includes a TextView, EditText, and the TextInputLayout legacy and updated styles side by side?

Reply:
Would you mind giving me the XML code you used in your screenshot?

Sidenote: what would the _bottom_ offset have to do with anything? The baseline only cares about what's above it AFAIK.

Reply:
#### @afohrman This is with current code, I really don't see what's wrong in terms of baseline:
![screenshot_20180906-195442](https://user-images.githubusercontent.com/9490724/45195997-0866ab80-b20f-11e8-90f1-5e6caa565b88.png)

Reply:
Cool, this looks great. I'll merge the change.

Thanks for contributing!

Reply:
Oh never mind, that screenshot was using an earlier version of the code. I've verified that it works and I'll merge the changes.

Thank you!

Reply:
Phew! 😌 Can't wait to see this finally go through!!! 😁👏🎉